xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/gallery/gallery.js

```diff
@@ -133,11 +133,18 @@ XWiki.Gallery = Class.create({
     }
     // Update only if it's a different image. Some browsers, e.g. Chrome, don't fire the load event if the image URL
     // doesn't change. Another trick would be to reset the src attribute before setting the actual URL (set to '').
-    if (this.currentImage.src != this.images[index].url) {
+    let imageData = this.images[index];
+    if (this.currentImage.src !== imageData.url) {
       this.currentImage.style.visibility = 'hidden';
       Element.addClassName(this.currentImage.parentNode, 'loading');
-      this.currentImage.title = this.images[index].title;
-      this.currentImage.src = this.images[index].url;
+      this.currentImage.title = imageData.title;
+      // If the alt is just the name of the file, we instead fall back on the human-readable currentImage translation.
+      if (!imageData.url.includes(imageData.alt)) {
+        this.currentImage.alt = imageData.alt;
+      } else {
+        this.currentImage.alt = '$escapetool.xml($services.localization.render("core.widgets.gallery.currentImage"))';
```

Comment:
Are you sure the xml is the right escaping in the middle of a js file?

Reply:
Also, code generation in the middle of a js file is not really advised, as it makes keeping the js in cache very difficult (among other issues).

Reply:
See https://extensions.xwiki.org/xwiki/bin/view/Extension/Localization/#HFromJavaScript how to access translations in JavaScript.